Sources/Sentry/SentryTracer.m

```diff
@@ -160,12 +170,23 @@ - (instancetype)initWithTransactionContext:(SentryTransactionContext *)transacti
         self.finishStatus = kSentrySpanStatusUndefined;
         self.idleTimeout = idleTimeout;
         self.dispatchQueueWrapper = dispatchQueueWrapper;
+
+        if (timerWrapper == nil) {
+            self.timerWrapper = [[SentryNSTimerWrapper alloc] init];
+        } else {
+            self.timerWrapper = timerWrapper;
+        }
```

Comment:
`h`: What's the point of passing the `timerWrapper` in as a dependency when we init it here? I think we could remove passing it in as a dependency, init it here and use `Dynamic` or a setter only visible internally to change the timerWrapper out for the `TestSentryNSTimerWrapper` for tests.

Reply:
We can't use Dynamic or a setter to change the timerWrapper, as it is already used in the initializer. So we need to pass it in. But I made the parameter optional so we don't need change a big amount of code, when we only need to change the timerWrapper in a test. Thus, we can either pass it in, or we create one here.

Reply:
I think we shouldn't initialize one here. Instead, I would vote for applying the same approach as for the `dispatchQueueWrapper`. If it's nil, it's nil, and we don't do anything.

Reply:
Uhm but we need a timer wrapper, so we need to initialize one here. The only reason why nil can be passed in, is because that saves a LOT of refactoring. Instead of every place that creates a SentryTracer object having to pass in a timerWrapper instance, we don't have to deal with any of that. Only the tests have to pass in a custom timerWrapper, everything else uses the default one which is created here.

What you are saying is not going to work.

Reply:
Then let's leave it. It saves us some refactoring, as you pointed out.